trieste/models/gpflow/sampler.py

```diff
@@ -348,84 +455,245 @@ def _prepare_theta_posterior_in_gram_space(self) -> tfp.distributions.Multivaria
+            (  # extract variational parameters
+                inducing_points,
+                q_mu,
+                q_sqrt,
+                whiten,
+            ) = self._model.get_inducing_variables()  # [M, d], [M, 1], [1, M, 1]
+            q_sqrt = q_sqrt[0, :, :]  # [M, M]
+            Kmm = self._model.get_kernel().K(inducing_points, inducing_points)  # [M, M]
+            Kmm += tf.eye(tf.shape(inducing_points)[0], dtype=Kmm.dtype) * DEFAULTS.JITTER
+        else:  # massage quantities from GP to look like variational parameters
+            internal_data = self._model.get_internal_data()
+            inducing_points = internal_data.query_points  # [M, d]
+            q_mu = self._model.get_internal_data().observations  # [M, 1]
+            q_sqrt = tf.eye(tf.shape(inducing_points)[0], dtype=tf.float64)  # [M, M]
+            q_sqrt = tf.math.sqrt(self._model.get_observation_noise()) * q_sqrt
+            whiten = False
+            Kmm = (
+                self._model.get_kernel().K(inducing_points, inducing_points) + q_sqrt ** 2
+            )  # [M, M]
+
+        tf.debugging.assert_shapes(
+            [
+                (inducing_points, ["M", "d"]),
+                (q_mu, ["M", "1"]),
+                (q_sqrt, ["M", "M"]),
+                (Kmm, ["M", "M"]),
+            ]
         )
 
-        self._theta_posterior = self._build_theta_posterior()  # recalculate weight distribution
+        def weight_sampler(batch_size: int) -> Tuple[TensorType, TensorType]:
 
-        trajectory.update(weight_distribution=self._theta_posterior)  # type: ignore
+            prior_weights = tf.random.normal(  # Non-RFF features will require scaling here
+                [self._num_features, batch_size], dtype=tf.float64
+            )  # [L, B]
 
-        return trajectory  # return trajectory with updated features and weight distribution
+            u_noise_sample = tf.matmul(
+                q_sqrt,  # [M, M]
+                tf.random.normal(
+                    (tf.shape(inducing_points)[0], batch_size), dtype=tf.float64
+                ),  # [ M, B]
+            )  # [M, B]
 
-    def resample_trajectory(self, trajectory: TrajectoryFunction) -> TrajectoryFunction:
+            u_sample = q_mu + u_noise_sample  # [M, B]
+            if whiten:
+                Luu = tf.linalg.cholesky(Kmm)  # [M, M]
+                u_sample = tf.matmul(Luu, u_sample)  # [M, B]
+
+            phi_Z = self._feature_functions(inducing_points)[:, : self._num_features]  # [M, B]
+            weight_space_prior_Z = phi_Z @ prior_weights  # [M, B]
+
+            diff = u_sample - weight_space_prior_Z  # [M, B]
+            v = compute_A_inv_b(Kmm, diff)  # [M, B]
+
+            tf.debugging.assert_shapes([(v, ["M", "B"]), (prior_weights, ["L", "B"])])
+
+            return tf.transpose(tf.concat([prior_weights, v], axis=0))  # [B, L + M]
+
+        return weight_sampler
+
+
+class ResampleableRandomFourierFeatureFunctions(RFF):  # type: ignore
```

Comment:
Including the specific type error you're ignoring is good practice (both to catch errors and for documentation):

```suggestion
class ResampleableRandomFourierFeatureFunctions(RFF):  # type: ignore[misc]
```